subsys/net/lib/sockets/sockets.c

```diff
@@ -98,6 +91,15 @@ int _impl_zsock_socket(int family, int type, int proto)
 	return fd;
 }
 
+int _impl_zsock_socket(int family, int type, int proto)
+{
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
```

Comment:
My intention was not to inject too much TLS realated code into `sockets.c`. But I can see that as there are more socket 'types' coming into Zephyr, this might be unavoidable. Will rethink this.

Reply:
Let's leave as is for now.